stormlog/telemetry_rollups.py

```
@@ -0,0 +1,902 @@
+def read_telemetry_rollups(path: str | Path) -> TelemetryRollupFile | None:
+    """Read a rollup sidecar if it exists and has a valid current shape."""
+
+    rollup_path = resolve_telemetry_rollup_path(path)
+    if not rollup_path.exists():
+        return None
+    try:
+        payload = json.loads(rollup_path.read_text(encoding="utf-8"))
+        if not isinstance(payload, Mapping):
+            return None
+        rollups = telemetry_rollup_file_from_dict(payload)
+    except Exception:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)